vertx-service-discovery-bridge-kubernetes/src/main/java/io/vertx/servicediscovery/kubernetes/KubernetesServiceImporter.java

```diff
@@ -61,16 +58,26 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(KubernetesServiceImporter.class.getName());
   public static final String KUBERNETES_UUID = "kubernetes.uuid";
 
+  private final Map<RecordKey, Record> records = new HashMap<>();
+
+  private ContextInternal context;
   private ServicePublisher publisher;
+  private String token;
   private String namespace;
-  private List<Record> records = new CopyOnWriteArrayList<>();
-  private WebClient client;
+  private HttpClient client;
+  private String lastResourceVersion;
+
+  private volatile boolean stop;
 
   private static final String OPENSHIFT_KUBERNETES_TOKEN_FILE = "/var/run/secrets/kubernetes.io/serviceaccount/token";
 
   @Override
-  public void start(Vertx vertx, ServicePublisher publisher, JsonObject configuration,
-                    Promise<Void> completion) {
+  public void start(Vertx vertx, ServicePublisher publisher, JsonObject configuration, Promise<Void> completion) {
+    context = (ContextInternal) vertx.getOrCreateContext();
```

Comment:
Nothing to do with that review, but it makes me think that somehow, somewhere prominent, we should document accessing `ContextInternal`, why and how.